src/libAtomVM/nifs.c

```diff
@@ -2544,9 +2544,30 @@ static term nif_erlang_list_to_binary_1(Context *ctx, int argc, term argv[])
     return bin_res;
 }
 
-static term nif_erlang_list_to_integer_1(Context *ctx, int argc, term argv[])
+static avm_int_t to_digit_index(avm_int_t character)
 {
-    UNUSED(argc);
+    if (character >= '0' && character <= '9') {
+        return character - '0';
+    } else if (character >= 'a' && character <= 'z') {
+        return character - 'a' + 10;
+    } else if (character >= 'A' && character <= 'Z') {
+        return character - 'A' + 10;
+    } else {
+        return -1;
+    }
+}
+
+static term nif_erlang_list_to_integer(Context *ctx, int argc, term argv[])
+{
+    avm_int_t base = 10;
+    if(argc == 2) {
```

Comment:
let's avoid formatting error: `if(argc == 2) {` -> `if (argc == 2) {`. We also have a `.clang-format` config file if you are looking for it.